Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -391,24 +394,30 @@ static OSStatus sampleCallback(CMBufferRef buffer, void* refcon)
         return;
     }
 
-    ++m_framesBeingDecoded;
+    ++m_compressedSampleQueueSize;
```

Comment:
Do we need this incrementing in main thread, or could we just update `++m_compressedSampleQueueSize` when we modify m_compressedSampleQueue in the lambda?

Reply:
yes, so that the SourceBufferAVFObjC doesn't potentially append hundreds of frames until m_compressedSampleQueueSize in increased (as until then VideoMediaSampleRenderer::isReadyForMoreMediaData() will return true)